File: performance_parallelism/src/repeat_adder.c
import sys
import time

# usage: python repeat_adder.py a b
a = int(sys.argv[1])
b = int(sys.argv[2])

start = time.time()

result = 0

# repeated addition: a * b = add a to itself b times
for _ in range(b):
    result += a

end = time.time()

print("Result:", result)
print("Time:", end - start)import sys
import time

# usage: python repeat_adder.py a b
a = int(sys.argv[1])
b = int(sys.argv[2])

start = time.time()

result = 0

# repeated addition: a * b = add a to itself b times
for _ in range(b):
    result += a

end = time.time()

print("Result:", result)
print("Time:", end - start)

File: performance_and_parallelism/repeat_adder.py
import sys

def main():
    try:
        in_arg = int(sys.argv[1])
        multiple = int(sys.argv[2])
    except:
        raise Exception("Usage: python repeat_adder.py [NUM] [NUM]")

    output = 0

    for i in range(in_arg):
        output += multiple

    print(output)

if __name__ == "__main__":
    main()
